feat(bpdm): post endpoint to upload business partner data using csv file

SujitMBRDI opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description
<!-- 
Please describe your PR: 
- What does this PR introduce? 
- Does it fix a bug? 
- Does it add a new feature?
- Is it enhancing documentation?
-->

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->
This pull request contains changes for new POST api endpoint on BPDM Gate service, which will accept the CSV file as input for upsetting business partner input data. This endpoint enable way for user to create/update business partner using file.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

SujitMBRDI commented:
@nicoprow,
In general, I tried covering almost all necessary changes. 
Just except,  REST API conformity with permissions about new resource and will also adjust default realm files for both helm chart and test.

SujitMBRDI commented:
> @nicoprow, In general, I tried covering almost all necessary changes. Just except, REST API conformity with permissions about new resource and will also adjust default realm files for both helm chart and test.

@nicoprow, I have adapted changes this changes. Kindly review the changes made.

nicoprow reviewed (changes requested):
Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below

SujitMBRDI commented:
> Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below

I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.

SujitMBRDI commented on bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/model/PartnerUploadFileRow.kt:
```diff
@@ -0,0 +1,214 @@
+
+data class PartnerUploadFileRow(
+
+    @CsvBindByName(column = "externalId")
```
Done

SujitMBRDI commented on bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/util/PartnerFileUtil.kt:
```diff
@@ -0,0 +1,149 @@
+                    )
+                ),
+                states = listOf(BusinessPartnerStateDto(
+                    validFrom = row.statesValidFrom?.let { LocalDateTime.parse(it, formatter) },
```
Adapted with new function

SujitMBRDI commented on bpdm-gate/src/test/kotlin/org/eclipse/tractusx/bpdm/gate/controller/PartnerUploadControllerIT.kt:
```diff
@@ -0,0 +1,170 @@
+)
+@ActiveProfiles("test-no-auth")
+@ContextConfiguration(initializers = [PostgreSQLContextInitializer::class])
+class PartnerUploadControllerIT @Autowired constructor(
```
Created new test which covers check for the expectation on created business partner input data.

SujitMBRDI commented on charts/bpdm/Chart.yaml:
```diff
@@ -22,7 +22,7 @@ apiVersion: v2
 description: A Helm chart for Kubernetes that deploys the BPDM applications
-version: 5.0.1
+version: 5.0.2
 appVersion: "6.0.1"
```
Done

SujitMBRDI commented on bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/response/PartnerUploadErrorResponse.kt:
```diff
@@ -0,0 +1,36 @@
+    @Schema(description = "HTTP status of the error response")
+    val status: HttpStatus,
+    @Schema(description = "List of error messages")
+    val error: List<String>,
```
As mentioned on sig release issue [700](https://github.com/eclipse-tractusx/sig-release/issues/700).
More information may come from Juila so, for now i haven't refactored this much.But yes in future we can refactor this. Keeping this for future scope, hope you agree with this.

SujitMBRDI commented on bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GatePartnerUploadApi.kt:
```diff
@@ -0,0 +1,66 @@
+                )]),
+        ]
+    )
+    @PostMapping("/input/partner-upload-process/upload-partner-csv", consumes = ["multipart/form-data"])
```
Done

nicoprow commented:
> > Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below
> 
> I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.

I the current app version is 6.0.1-SNAPSHOT. Since this pull request adds a new feature to the app, we need to increase the SNAPSHOT version to 6.1.0-SNAPSHOT. Likewise, we need to increase the chart's SNAPSHOT version to 5.1.0-SNAPSHOT as well as update its appVersion to the 6.1.0-SNAPSHOT.

SujitMBRDI commented:
> > > Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below
> > 
> > 
> > I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.
> 
> @SujitMBRDI I the current app version is 6.0.1-SNAPSHOT. Since this pull request adds a new feature to the app, we need to increase the SNAPSHOT version to 6.1.0-SNAPSHOT. Likewise, we need to increase the chart's SNAPSHOT version to 5.1.0-SNAPSHOT as well as update its appVersion to the 6.1.0-SNAPSHOT.

@nicoprow, Done. I have adapted changes.

nicoprow reviewed (approved):
Looks good to me
